### PR TITLE
Added ability to edit tasks

### PR DESCRIPTION
**Pull request by RyhorKrs**

### CODE REVIEW

File: src/modules/app/tasks/add-task-form/add-task-form.component.html [hunk 1]
```diff
@@ -11,6 +11,20 @@ <h1 class='title'>{{ 'titles.tasks-page' | translate }}</h1>
     <input matInput type="text"  placeholder="{{ 'placeholders.description' | translate }}" formControlName="taskDescription">
     <mat-error *ngIf="addTaskForm.get('taskDescription').hasError('required')">{{ 'errors.required' | translate }}</mat-error>
   </mat-form-field>
+
```
Comment (AliakseiKliashchou) [inline src/modules/app/tasks/add-task-form/add-task-form.component.html:14]: uselesss

File: src/modules/app/tasks/add-task-form/add-task-form.component.html [hunk 2]
```diff
@@ -11,6 +11,20 @@ <h1 class='title'>{{ 'titles.tasks-page' | translate }}</h1>
...
+    <mat-form-field class="example-full-width datepicker" appearance="fill">
+      <mat-label>{{ 'form-field-label.chose-deadline' | translate }}</mat-label>
+      <input autocomplete="none" matInput [matDatepicker]="picker" formControlName="taskDate">
+      <mat-datepicker-toggle matSuffix [for]="picker">
+        <mat-icon matDatepickerToggleIcon>keyboard_arrow_down</mat-icon>
+      </mat-datepicker-toggle>
+      <mat-datepicker #picker></mat-datepicker>
+      <mat-error *ngIf="addTaskForm.get('taskDate').hasError('required')">{{ 'errors.required' | translate }}</mat-error>
+    </mat-form-field>
+
```
Comment (AliakseiKliashchou) [inline src/modules/app/tasks/add-task-form/add-task-form.component.html:25]: uselesss

File: src/modules/app/tasks/tasks.component.scss [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  mat-card {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    max-width: 600px;
+  }
+
+  mat-card-title {
+    margin-bottom: 30px;
+  }
```
Comment (AliakseiKliashchou) [inline src/modules/app/tasks/tasks.component.scss:22]: The better way is to override the library via classes

File: src/modules/app/tasks/tasks.component.ts [hunk 4]
```diff
@@ -13,6 +13,22 @@ export class TasksComponent implements OnInit {
   public error: string = '';
   public user: any = {};
   public tasks: Task[] = [];
+  public newTasks: Task[] = [];
+  public inProcessTasks: Task[] = [];
+  public doneTasks: Task[] = [];
+  public currentDelTask: any = {};
```
Comment (AliakseiKliashchou) [inline src/modules/app/tasks/tasks.component.ts:19]: wierd

File: src/modules/app/tasks/tasks.component.ts [hunk 5]
```diff
@@ -13,6 +13,22 @@ export class TasksComponent implements OnInit {
   public error: string = '';
   public user: any = {};
```
Comment (AliakseiKliashchou) [inline src/modules/app/tasks/tasks.component.ts:14]: wierd

File: src/modules/app/tasks/tasks.component.ts [hunk 6]
```diff
@@ -29,12 +54,22 @@ export class TasksComponent implements OnInit {
...
         task.id = key;
         this.tasks.push(task);
+
+        if(task.taskCategory === 'new') {
+          this.newTasks.push(task);
+        } else if (task.taskCategory === 'in-process') {
+          this.inProcessTasks.push(task);
+        } else if (task.taskCategory === 'done') {
+          this.doneTasks.push(task);
+        }
```
Comment (AliakseiKliashchou) [inline src/modules/app/tasks/tasks.component.ts:72]: switch?

File: src/modules/app/tasks/tasks.component.ts [hunk 7]
```diff
@@ -44,16 +79,105 @@ export class TasksComponent implements OnInit {
...
+    document.body.style.cssText = `
+        position: fixed;
+        top: ${-window.scrollY}px;
+        left: 0;
+        width: 100%;
+        height: 100vh;
+        overflow: hidden;
+        padding-right: ${scrollWidth}px;
+    `;
+  }
```
Comment (AliakseiKliashchou) [inline src/modules/app/tasks/tasks.component.ts:119]: what is that?

File: src/modules/app/tasks/tasks.component.ts [hunk 8]
```diff
@@ -44,16 +79,105 @@ export class TasksComponent implements OnInit {
...
+    window.scroll({
+        top: this.topWindowScroll,
+    });
+  }
+
+  public preEditTask(task: Task): void {
+    this.showEditTaskModal = true;
+    this.currentEditTask = task;
+
+    let validDate = `${task.taskDate.substr(3,2)}.${task.taskDate.substr(0,2)}.${task.taskDate.substr(6,4)}`;
```
Comment (AliakseiKliashchou) [inline src/modules/app/tasks/tasks.component.ts:135]: Declare variables in the top of block

File: src/modules/app/tasks/tasks.component.ts [hunk 9]
```diff
@@ -44,16 +79,105 @@ export class TasksComponent implements OnInit {
...
+    };
+
+    this.fbTasksService.editTaskInDb(editedTask, JSON.parse(localStorage.uid), this.currentEditTask.id).subscribe(() => {
+      this.tasks = [];
+      this.newTasks = [];
+      this.inProcessTasks = [];
+      this.doneTasks = [];
+      this.getTasksContent();
+      this.hideEditModal();
+    }, err => {
```
Comment (AliakseiKliashchou) [inline src/modules/app/tasks/tasks.component.ts:161]: new line

File: src/modules/app/tasks/add-task-form/add-task-form.component.html (hunk 1 above)
Comment (RyhorKrs) [reply src/modules/app/tasks/add-task-form/add-task-form.component.html:14]: I can remove lines if it necessary, but I think code reads better with spaces.

File: src/modules/app/tasks/add-task-form/add-task-form.component.html (hunk 2 above)
Comment (RyhorKrs) [reply src/modules/app/tasks/add-task-form/add-task-form.component.html:25]: I can remove lines if it necessary, but I think code reads better with spaces.

File: src/modules/app/tasks/tasks.component.ts (hunk 4 above)
Comment (RyhorKrs) [reply src/modules/app/tasks/tasks.component.ts:19]: Unfortunately, these variables play important role in the logic of my code an i have no idea to way how remove them.

File: src/modules/app/tasks/tasks.component.ts (hunk 7 above)
Comment (RyhorKrs) [reply src/modules/app/tasks/tasks.component.ts:119]: This method adds padding to the browser window and when the modal window is opened, the browser does not move.

File: src/modules/app/tasks/add-task-form/add-task-form.component.html (hunk 1 above)
Comment (AliakseiKliashchou) [reply src/modules/app/tasks/add-task-form/add-task-form.component.html:14]: ok

File: src/modules/app/tasks/tasks.component.ts [hunk 10]
```diff
@@ -79,98 +61,40 @@ export class TasksComponent implements OnInit {
...
-    window.scroll({
-        top: this.topWindowScroll,
+    editDialogRef.afterClosed().subscribe(result => {
+      if (result) {
+        this.fbTasksService.editTaskInDb(result, JSON.parse(localStorage.uid), this.currentEditTask.id).subscribe(() => {
+          this.tasks = [];
+          this.newTasks = [];
+          this.inProcessTasks = [];
+          this.doneTasks = [];
+          this.getTasksContent();
```
Comment (AliakseiKliashchou) [inline src/modules/app/tasks/tasks.component.ts:93]: u can move resets to getTasksContent

**Review by AliakseiKliashchou — APPROVED** [review]

**Pull request merged**